easybuild/easyconfigs/f/FreeFem++/FreeFem++-3.58-foss-2017b-downloaded-deps.eb

```diff
@@ -0,0 +1,45 @@
+easyblock = 'ConfigureMake'
+
+name = 'FreeFem++'
+version = '3.58'
+versionsuffix = '-downloaded-deps'
+
+homepage = 'http://www.freefem.org/'
+description = """ FreeFem++ is a partial differential equation solver. It has its own language.
+freefem scripts can solve multiphysics non linear systems in 2D and 3D. """
+
+toolchain = {'name': 'foss', 'version': '2017b'}
+toolchainopts = {'usempi': True, 'pic': True}
+
+source_urls = ['http://www.freefem.org/ff++/ftp/']
+sources = ['%(namelower)s-3.58.tar.gz']
+checksums = ['e205c0a89fe8edc876d97b8a6ba43825b83c7f8dbdd49fdf972fa1f089b62743']
+
+builddependencies = [('Autotools', '20170619')]
+
+dependencies = [
+    ('HDF5', '1.10.1'),
+    ('GSL', '2.4'),
+    ('freeglut', '3.0.0'),
+]
+
+preconfigopts = "autoreconf -i &&"
+
+configopts = '--with-hdf5="$EBROOTHDF5"/bin '
+configopts += '--with-gsl-include="$EBROOTGSL"/include --with-gsl-ldflags="$EBROOTGSL"/lib '
+configopts += '--with-lapack="$LIBLAPACK" '
+
+# dependencies are heavily patched by FreeFem++, we therefore let it download and install them itself
+prebuildopts = 'download/getall -a -o ScaLAPACK,ARPACK,freeYams,Gmm++,Hips,Ipopt,METIS,ParMETIS,MMG3D,mshmet,MUMPS,'
+prebuildopts += 'NLopt,pARMS,PaStiX,Scotch,SuiteSparse,SuperLU_DIST,SuperLU,TetGen,PETSc,SLEPc,hpddm &&'
```

Comment:
@smoors Do they patch *all* of these?

Can we download these via `sources` and feed them in somehow (so we control which versions are used)?